pkg/controllers/management/drivers/nodedriver/machine_driver.go

```diff
@@ -179,7 +181,17 @@ func (m *Lifecycle) download(obj *v3.NodeDriver) (*v3.NodeDriver, error) {
 		return obj, errs.Wrap(err, "failed JSON in addUIHintsAnno")
 	}
 
-	flags, err := getCreateFlagsForDriver(driverName)
+	var flags []cli.Flag
+	if driverName == "gke" {
+		flags = []cli.Flag{
+			&cli.StringFlag{
+				Name: "gke-credential",
+			},
+		}
```

Comment:
I believe this is the only place the driver is called here. This is just to get the flags for creation.

The name of the flag should be `google-auth-encoded-json` to make it compatible with machine (in case we add more general support).

Reply:
If the changes do get merged into `rancher-machine`, then this block can be removed.

Reply:
Where does the string `google-auth-encoded-json` come from? I used `credential` because that's currently how kontainer-engine identifies it

https://github.com/rancher/kontainer-engine/blob/89626b028c6a0cb1eeb399c9e7dc5404286efbb4/drivers/gke/gke_driver.go#L323 

https://github.com/rancher/kontainer-engine/blob/89626b028c6a0cb1eeb399c9e7dc5404286efbb4/drivers/gke/gke_driver.go#L846